app/js/components/manageApps/ManageApps.jsx

```diff
@@ -458,6 +470,91 @@ export default class ManageApps extends React.Component {
     };
   }
 
+  handleUpgrade(addon, newVersion, addonUid) {
+    this.setState({
+      openUpgradeConfirmation: true,
+      upgradeAddon: addon,
+      upgradeVersion: newVersion,
+      upgrading: true,
+    });
+
+    axios.get(`https://addons.openmrs.org/api/v1//addon/${addonUid}`)
+      .then(response => {
+        const newAddon = {
+          appDetails: response.data,
+          downloadUri: response.data.versions[0].downloadUri,
+        }
+        this.setState({
```

Comment:
According to the [AirBnb JavaScript Style guide](https://github.com/airbnb/javascript#es6-object-concise), this should be `this. setState({ newAddon })`